agent/src/dispatcher/mirror_mode_dispatcher.rs

```diff
@@ -201,12 +201,215 @@ impl MirrorModeDispatcherListener {
 }
 
 pub(super) struct MirrorPipeline {
-    handlers: Vec<PacketHandler>,
+    pub handlers: Vec<PacketHandler>,
 }
 
 pub(super) struct LastTimestamps {
+    pub if_index: isize,
+    pub last_timestamp: Duration,
+}
+
+pub fn get_key(
+    vm_mac_set: &Arc<Mutex<HashMap<u32, MacAddr>>>,
+    overlay_packet: &[u8],
+    tunnel_info: TunnelInfo,
+) -> (u32, u32, u32, u32) {
+    let (da_key, sa_key) =
+        if tunnel_info.tier == 0 && overlay_packet.len() >= super::L2_MAC_ADDR_OFFSET {
+            let mut da_mac: [u8; 6] = [0; 6];
+            let mut sa_mac: [u8; 6] = [0; 6];
+            da_mac.copy_from_slice(&overlay_packet[..6]);
+            sa_mac.copy_from_slice(&overlay_packet[6..12]);
+            (
+                MacAddr::from(da_mac).to_lower_32b(),
```

Comment:
`MacAddr` 实现了 `TryFrom<[u8]>`

```suggestion
                MacAddr::try_from(&overlay_packet[..6]).unwrap().to_lower_32b(),
```